File: src/persona_composer/hashing.py
"""Hashing helpers."""

from __future__ import annotations

import hashlib
from pathlib import Path


def file_hash(path: Path) -> str:
    digest = hashlib.sha256(path.read_bytes()).hexdigest()
    return digest[:12]


def content_hash(content: str | bytes) -> str:
    data = content if isinstance(content, bytes) else content.encode("utf-8")
    return hashlib.sha256(data).hexdigest()[:12]

File: ts/src/hashing.ts
import { createHash } from "node:crypto";
import { readFileSync } from "node:fs";

export function fileHash(path: string): string {
  const digest = createHash("sha256").update(readFileSync(path)).digest("hex");
  return digest.slice(0, 12);
}

export function contentHash(content: string | Buffer): string {
  const digest = createHash("sha256").update(content).digest("hex");
  return digest.slice(0, 12);
}
